api/registrations/serializers.py

```diff
@@ -481,20 +482,71 @@ class Meta:
 
 class RegistrationCreateSerializer(RegistrationSerializer):
     """
-    Overrides RegistrationSerializer to add draft_registration, registration_choice, and lift_embargo fields
+    Overrides RegistrationSerializer to add draft_registration, registration_choice, and lift_embargo fields -
     """
-    draft_registration = ser.CharField(write_only=True)
-    registration_choice = ser.ChoiceField(write_only=True, choices=['immediate', 'embargo'])
+
+    def expect_cleaner_attributes(self, request):
+        return StrictVersion(getattr(request, 'version', '2.0')) >= StrictVersion(CREATE_REGISTRATION_FIELD_CHANGE_VERSION)
+
+    def __init__(self, *args, **kwargs):
+        super(RegistrationCreateSerializer, self).__init__(*args, **kwargs)
+        request = kwargs['context']['request']
+        # required fields defined here for the different versions
+        if self.expect_cleaner_attributes(request):
+            self.fields['draft_registration_id'] = ser.CharField(write_only=True)
+        else:
+            self.fields['draft_registration'] = ser.CharField(write_only=True)
+            self.fields['registration_choice'] = ser.ChoiceField(write_only=True, choices=['immediate', 'embargo'])
+
+    # For newer versions
+    embargo_end_date = VersionedDateTimeField(write_only=True, default=None, input_formats=['%Y-%m-%dT%H:%M:%S'])
+    included_node_ids = ser.ListField(write_only=True, required=False)
```

Comment:
Versioned it out, since some of these fields are required on POST, so it was easier to make one set required pre-2.18, and another set required after 2.18